packages/SwingSet/src/kernel/kernel.js

```diff
@@ -1202,6 +1202,19 @@ export default function buildKernel(
     return results;
   }
 
+  function getCrankVatForSlog(message) {
+    if (message.type === 'send') {
+      const { type } = parseKernelSlot(message.target);
+      if (type !== 'promise') {
+        return kernelKeeper.ownerOfKernelObject(message.target);
```

Comment:
I think I've figured out how to remove the extra work from the kernel.  Or rather, I always knew how to do it but it required adding a bunch of gnarly lookahead gymnastics to the way the slogulator consumes the slog file and produces its output. However, in order to accommodate reporting some of the timing data, which I'm finding it really needs to report, it looks like it'll need to do many those gymnastics anyway, which caused me to examine more deeply how hard it would be.  Based on that deeper analysis I don't think it will actually be nearly as nasty to implement as I had originally feared.  So I'll shortly be pushing another commit that will take out the kernel changes while adding some timing reporting stuff to the slogulator itself.